File: 0898-transpose-matrix/0898-transpose-matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rows = len(matrix)
        cols = len(matrix[0])
        res = []

        for i in range(cols):
            new = []
            for j in range(rows):
                new.append(matrix[j][i])
            res.append(new)
        
        return res

File: 0898-transpose-matrix/0898-transpose-matrix.js
/**
 * @param {number[][]} matrix
 * @return {number[][]}
 */
var transpose = function(matrix) {
    const result = [];

    for (let i = 0; i < matrix[0].length; i++) {
        const newRow = [];
        for (let j = 0; j < matrix.length; j++) {
            newRow.push(matrix[j][i]);
        }
        result.push(newRow);
    }

    return result;
};